docs/Advanced/SQL Searches/SQL WHERE Clause.md

```diff
@@ -0,0 +1,44 @@
+---
+publish: true
+---
+
+# WHERE Clause
+
+<span class="related-pages">#advanced/sql-search</span>
+
+## Operators in The WHERE Clause
+
+The following basic operators can be used in the `WHERE` clause. More can be found on the [SQL Operators](./SQL Operators) page.
+
+| Operator | Description                                                                 | Example |
+| -------- | --------------------------------------------------------------------------- | ------- |
+| =        | Equal                                                                       |         |
+| >        | Greater than                                                                |         |
+| <        | Less than                                                                   |         |
+| >=       | Greater than or equal                                                       |         |
+| <=       | Less than or equal                                                          |         |
+| <>       | Not equal. Note: In some versions of SQL this operator may be written as != |         |
+| BETWEEN  | Between a certain range                                                     |         |
+| LIKE     | Search for a pattern                                                        |         |
+| IN       | To specify multiple possible values for a column                            |         |
+
+## The SQL AND, OR and NOT Operators
+
+The `WHERE` clause can be combined with `AND`, `OR`, and `NOT` operators.
+
+The `AND` and `OR` operators are used to filter records based on more than one condition:
+
+- The `AND` operator displays a record if all the conditions separated by `AND` are TRUE.
+- The `OR` operator displays a record if any of the conditions separated by `OR` is TRUE.
+
+The `NOT` operator displays a record if the condition(s) is NOT TRUE.
+
+You can also combine the `AND`, `OR` and `NOT` operators.
+
+The following SQL statement selects all tasks where priority is "2" AND indicator must be "!" OR "?" (use parenthesis to form complex expressions)
+
+````text
+```task-sql
+WHERE priority = '2' AND (status->symbol='!' OR status->symbol='?')
+```
+````
```

Comment:
I picked this one at random to try out.

When I first pasted it in to Tasks, it just displayed the query text.

Eventually I noticed that `task-sql` was spelled wrong.

When I fixed that, I got:

![image](https://user-images.githubusercontent.com/4840096/231837013-c4e494f1-4f0e-4cb6-aea2-d340a162fca9.png)

When I replaced the NBSP in the query with spaces, it worked:

~~~text
```tasks-sql
WHERE priority = '2' AND (status->symbol='!' OR status->symbol='?')
```

- [!] #task Low 🔽
- [!] #task Normal
- [?] #task Medium 🔼
- [?] #task High 
~~~

With `S - Checkboxes` enabled, it looked like this:

![image](https://user-images.githubusercontent.com/4840096/231837941-a82b8621-ad39-497f-a234-3a67fc4c3862.png)


![image](https://user-images.githubusercontent.com/4840096/231837851-625f968b-c7ed-43e6-8a6d-1a45c2f8e0dd.png)

Reply:
Actually, if you could systematically replace all the NBSP with space, that would help.

I tend to see NBSP when I copy text from web pages...

Was any of the general descriptive text about SQL copied from any website? If so, which one please?

Asking because I want to be confident we don't break any copyright/licensing terms.

Reply:
Some of the standard details about the comparisons that can be used were taken from the AlaSQL site which is under the same MIT license. The rest I generated. There is a link to the AlaSQL site on that page from memory.

I will look for the NBSP instances and remove them where I find them. 

I thought I caught all the misspellings of the task(s)-sql. Will do a search and replace in VSCode and then validate in the vault. Same for the NBSP instances.

Reply:
> Some of the standard details about the comparisons that can be used were taken from the AlaSQL site which is under the same MIT license. The rest I generated. There is a link to the AlaSQL site on that page from memory.
> 
> I will look for the NBSP instances and remove them where I find them.
> 
> I thought I caught all the misspellings of the task(s)-sql. Will do a search and replace in VSCode and then validate in the vault. Same for the NBSP instances.

Fantastic. Thank you.

Reply:
Did a global search and replace for the NBSP character, all instances have been replaced.

Reply:
tasks-sql is now consistent across the files.